### PR TITLE
handle gregor show_tracker_popup notifications in the client

### PR DESCRIPTION
**Pull request by oconnor663**
This is my first time using `UIRouter`, so there's a good chance I've put this together incorrectly. Also please let me know if there's a better way to kick off a track from the service, other than using `TrackEngine`. I was able to see a tracker popup on my machine using `gregor_inject.iced`.

r? @maxtaco @patrickxb

### CODE REVIEW

File: go/service/gregor.go [hunk 1]
```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
...
+	identifyUI, err := g.G().UIRouter.GetIdentifyUI()
+	if err != nil {
+		g.G().Log.Error("failed to get IdentifyUI", err)
+		return err
+	}
+	if identifyUI == nil {
+		g.G().Log.Error("got nil IdentifyUI")
+		return errors.New("got nil IdentifyUI")
+	}
+	secretUI, err := g.G().UIRouter.GetSecretUI(0 /* unused? */)
```
Comment (patrickxb) [inline go/service/gregor.go]: 0 is special for electron, so it matters...but is correct in this instance.

File: go/service/gregor.go [hunk 2]
```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
...
+			}
+			if category == "show_tracker_popup" {
+				return g.handleShowTrackerPopup(ctx, item)
+			}
+			g.G().Log.Errorf("Unrecognized item category: %s", item.Category())
+		}
+
+		dismissal := update.Dismissal()
+		if dismissal != nil {
+			g.G().Log.Debug("gregor handler: dismissal!!!")
```
Comment (patrickxb) [inline go/service/gregor.go]: I'm assuming this a different ticket to handle this path?

File: go/service/gregor.go [hunk 3]
```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
...
+
+	return fmt.Errorf("gregor handler: both in-band and out-of-band message nil")
+}
+
+func (g *gregorHandler) handleInBandMessage(ctx context.Context, ibm gregor.InBandMessage) error {
+	g.G().Log.Debug("gregor handler: handleInBand: %+v", ibm)
+
+	sync := ibm.ToStateSyncMessage()
+	if sync != nil {
+		g.G().Log.Debug("gregor handler: state sync message")
```
Comment (patrickxb) [inline go/service/gregor.go]: return here?

File: go/service/gregor.go [hunk 4]
```diff
@@ -91,11 +92,109 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
...
+	uid, err := body.AtPath("uid").GetString()
+	if err != nil {
+		g.G().Log.Error("failed to extract uid", err)
+		return err
+	}
+
+	// Load the user with that UID, to get a username.
+	// TODO: Should we load by username@service instead? I worry that might not
+	// be unique...
+	user, err := libkb.LoadUser(libkb.LoadUserArg{UID: keybase1.UID(uid)})
```
Comment (patrickxb) [inline go/service/gregor.go]: uid seems good to me.

Can you use `libkb.NewLoadUserByUIDArg(...)` so it it contextified?

**Comment by patrickxb** [comment]
I had a few comments, but looks good.

Can we add a test in `gregor_test.go`?  It would be a bit of a pain to test that it gets all the way to a test identify ui, but some test would be nice.  

Perhaps we should have support in systests for gregor messages?

File: go/service/gregor.go (hunk 2 above)
Comment (oconnor663) [inline go/service/gregor.go]: yep: https://keybase.atlassian.net/browse/CORE-2977

File: go/service/gregor.go (hunk 3 above)
Comment (oconnor663) [inline go/service/gregor.go]: Are we certain we will never have a message that has both sync and update? I guess we are?

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**